### PR TITLE
Add Inline Comment

### LINKED ISSUE #81: Consider line comment
A line comment that is different from a block comment would be nice

1. just typing something like // at the start of the line is much more convenient than putting something at the front and the end 

2. Many editors have the concept of line and block comment - so my intellij plugin for example works a lot better with both.

The existing block comment is great `/* */`  

maybe a line comment could be `//`

### PR DESCRIPTION
**Pull request by wisn**
## Summary

Add inline comment support.
This fixes #81.

### CODE REVIEW

File: src/ast/comment.cr [hunk 1]
```diff
@@ -1,9 +1,10 @@
 module Mint
   class Ast
     class Comment < Node
-      getter value
+      getter value, type
```
Comment (gdotdesign) [inline src/ast/comment.cr:4]: I would use an enum for this:

```
enum CommentType
  Inline
  Block
end
```

**Review by gdotdesign — APPROVED** [review]

File: src/ast/comment.cr (hunk 1 above)
Comment (wisn) [reply src/ast/comment.cr:4]: @gdotdesign sure

File: src/ast/comment.cr [hunk 2]
```diff
@@ -1,9 +1,15 @@
 module Mint
   class Ast
+    enum CommentType
+      Inline
+      Block
+    end
```
Comment (Sija) [inline src/ast/comment.cr:6]: Please move into a `Comment::Type`.

File: src/ast/comment.cr [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+
     class Comment < Node
-      getter value
+      getter value, type
 
       def initialize(@value : String,
+                     @type : CommentType,
```
Comment (Sija) [inline src/ast/comment.cr:12]:
```suggestion
                     @type : Type,
```

File: src/formatters/comment.cr [hunk 4]
```diff
@@ -7,10 +7,18 @@ module Mint
           .remove_leading_whitespace
           .rstrip
 
-      if replace_skipped(value).includes?('\n')
-        "/*\n#{value}\n*/"
+      if node.type == Ast::CommentType::Block
```
Comment (Sija) [inline src/formatters/comment.cr:10]:
```suggestion
      if node.type.block?
```

File: src/parsers/comment.cr [hunk 5]
```diff
@@ -2,17 +2,32 @@ module Mint
...
+            gather { consume_while((!(keyword_ahead "*/") || char == '\0') && !eof?) }.to_s
+
+          keyword "*/"
+        else
+          keyword "//"
+
+          value =
+            gather { consume_while(!((keyword_ahead "\n") || char == '\0') && !eof?) }.to_s
+          type = Ast::CommentType::Inline
+        end
```
Comment (Sija) [inline src/parsers/comment.cr:23]:
```suggestion
        value = nil

        if keyword_ahead "/*"
          keyword "/*"

          type = Ast::CommentType::Block
          value =
            gather { consume_while((!(keyword_ahead "*/") || char == '\0') && !eof?) }.to_s

          keyword "*/"
        else
          keyword "//"

          type = Ast::CommentType::Inline
          value =
            gather { consume_while(!((keyword_ahead "\n") || char == '\0') && !eof?) }.to_s
        end
```

**Review by Sija — CHANGES_REQUESTED** [review]

**Review by Sija — APPROVED** [review]
LGTM, thanks @wisn! 🚀

**Pull request merged**